waffle_utils/dataset/dataset.py

```diff
@@ -791,3 +873,136 @@ def _export(images: list[Image], set_name: str, export_dir: Path):
+            annotation_id (int): The starting annotation ID to be used for adding new annotations.
+
+        Returns:
+            int: The updated annotation ID after processing all annotations in the file.
+        """
+        # Load and split the YOLO annotation file into lines
+        yolo_txt = io.load_txt(yolo_txt_file)
+        yolo_txt_lines = yolo_txt.strip().splitlines()
+
+        # Process each line in the YOLO annotation file
+        for yolo_txt_line in yolo_txt_lines:
+            # Parse the class_id and vertices from the annotation line
+            class_id, *vertices = map(float, yolo_txt_line.split())
+            class_id = int(class_id)
+
+            # Scale the x and y coordinates of the vertices
+            scaled_x_coords = [
+                int(np.round(x * image_width)) for x in vertices[0::2]
+            ]
+            scaled_y_coords = [
+                int(np.round(y * image_height)) for y in vertices[1::2]
+            ]
+
+            # Calculate the bounding box coordinates and dimensions
+            xmin, xmax, ymin, ymax = (
+                min(scaled_x_coords),
+                max(scaled_x_coords),
+                min(scaled_y_coords),
+                max(scaled_y_coords),
+            )
+            bbox_width = int(np.round(xmax - xmin))
+            bbox_height = int(np.round(ymax - ymin))
+
+            # Create the bounding box and segmentation data
+            bbox = [xmin, ymin, bbox_width, bbox_height]
+            segmentation = polygon_to_rle(
+                image_width, image_height, scaled_x_coords, scaled_y_coords
+            )
+            area = bbox_width * bbox_height
+
+            # Check if the values are integers
+            assert isinstance(annotation_id, int)
+            assert isinstance(image_id, int)
+            assert isinstance(class_id, int)
+            assert isinstance(bbox, list) and all(
+                isinstance(x, int) for x in bbox
+            )
+            assert isinstance(image_width, int)
+            assert isinstance(image_height, int)
+            assert isinstance(segmentation, list) and all(
+                isinstance(x, int) for x in segmentation
+            )
+            assert isinstance(area, int)
+
+            # Create the annotation dictionary and add it to the dataset
+            annotation = Annotation.from_dict(
+                {
+                    "annotation_id": annotation_id,
+                    "image_id": image_id,
+                    "category_id": class_id + 1,
+                    "bbox": bbox,
+                    "segmentation": {
+                        "size": [image_width, image_height],
+                        "counts": segmentation,
+                    },
+                    "area": area,
+                }
+            )
+
+            ds.add_annotations([annotation])
+            annotation_id += 1
+
+        # Return the updated annotation_id
+        return annotation_id
```

Comment:
These function shouldn't be class method. As we talked about (before)[https://github.com/snuailab/waffle_utils/pull/52#discussion_r1147249190], these should be moved to the `from_yolo` function.